stack/language_template.go

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Alex Ellis 2017. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package stack
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	yaml "gopkg.in/yaml.v2"
+)
+
+func ParseYAMLForLanguageTemplate(file string) (*LanguageTemplate, error) {
+	if object, err := ParseYAML(
+		file,
+		iParseYAMLDataForLanguageTemplate,
+	); err != nil {
+		return nil, err
+	} else {
+		return object.(*LanguageTemplate), nil
+	}
+}
+
+// ParseYAMLDataForLanguageTemplate parses YAML data into language template
+func ParseYAMLDataForLanguageTemplate(fileData []byte, args ...string) (*LanguageTemplate, error) {
+	if object, err := iParseYAMLDataForLanguageTemplate(fileData, args...); err != nil {
+		return nil, err
+	} else {
+		return object.(*LanguageTemplate), nil
+	}
+}
+
+// iParseYAMLDataForLanguageTemplate parses YAML data into language template
+// Use the alias ParseYAMLDataForLanguageTemplate
+func iParseYAMLDataForLanguageTemplate(fileData []byte, args ...string) (interface{}, error) {
```

Comment:
What's the "i"-prefix convention?

Reply:
Because it returns an interface.

Reply:
Actually I'm going to try doing this without the interface approach.

Reply:
Actually "i" was for "internal", I wanted not to expose it and use the public method to do the assertion on return type.

That's why ParseYAMLDataForLanguageTemplate calls iParseYAMLDataForLanguageTemplate.

This re-factoring is because I wanted to extract ParseYAMLFile to ParseYAML, since the part of parsing can be re-used for both LanguageTemplate & Stack

Reply:
I am thinking we will add this re-factoring in a future PR, after this one is merged. The plan for this PR is to add minimal functionality.